data_model/derive/src/lib.rs

```diff
@@ -33,12 +19,19 @@ impl syn::parse::Parse for Items {
 /// Additionally, this macro will rewrite private items as public when `transparent_api` is active.
 /// If an item should remain private regardless of consumer library, just don't wrap it in this macro.
 ///
+/// Items private fields will be marked as `pub(super)`, so that it will be possible to write
+/// proper `impl`s for such items.
```

Comment:
I don't see a reason why you would make private fields be anything else. IMO a private field should remain private

Reply:
Making them `pub(super)` enables to write `impl`s in parent modules. So that we try to keep only struct and enum declarations inside `model` macro. Do you think we should move impl into `model` as well?

Reply:
why doesn't the developer just make it `pub(super)` then? If I make a field private I would expect it to stay private. Either don't allow for a field to be private or leave it private. It's ok to hide away struct internals, but but I would never reveal additional details about the struct.

Reply:
Okay, agree, it will be more explicit